ncm-afsclt/src/main/perl/afsclt.pm

```diff
@@ -2,997 +2,306 @@
 # ${developer-info}
 # ${author-info}
 
-# Coding style: emulate <TAB> characters with 4 spaces, thanks!
-################################################################################
-
 package NCM::Component::afsclt;
 
-#
-# a few standard statements, mandatory for all components
-#
-
 use strict;
+use Readonly;
 use NCM::Component;
 use NCM::Check;
 use LWP::Simple;
+use CAF::FileWriter;
+use CAF::FileEditor;
+use CAF::Process;
 
 use vars qw(@ISA $EC);
 @ISA = qw(NCM::Component);
-$EC=LC::Exception::Context->new->will_store_all;
-
+$EC  = LC::Exception::Context->new->will_store_all;
 $NCM::Component::afsclt::NoActionSupported = 1;
 
-# prevent authconfig from trying to launch in X11 mode
-delete($ENV{"DISPLAY"});
-
-#*** The following gives the OS name using the LC perl modules.
-#*** In example, $OSname can be 'Linux' or 'Solaris'
-#*** It's equivalent to perl's variable $^O (it gives the OS in lower-case)
-use LC::Sysinfo;
-my $OSname=LC::Sysinfo::os()->name;
-my $authconfig='/usr/bin/authconfig';
-my $iptables = '/etc/sysconfig/iptables';
-my $sysconfig_afs = '/etc/sysconfig/afs';
-my $afs_cacheinfo = '/usr/vice/etc/cacheinfo';
-my $localcelldb = "/usr/vice/etc/CellServDB";
-
-my $mypath='/software/components/afsclt'; # this could be rather in /system/filesystems/remote ?
-my $compname = "NCM-afsclt";
+Readonly my $AFS_CACHEINFO => '/usr/vice/etc/cacheinfo';
+Readonly my $LOCALCELLDB   => '/usr/vice/etc/CellServDB';
+Readonly my $THISCELL      => '/usr/vice/etc/ThisCell';
+Readonly my $THESECELLS    => '/usr/vice/etc/TheseCells';
+Readonly my $AFSD_ARGS     => '/etc/afsd.args';
 
-#***    In Solaris the fs executable at /usr/afsws/bin
-#*** so we put that directory first in the path.
-if ($OSname=~ /Solaris/) {$ENV{'PATH'}="/usr/afsws/bin:$ENV{'PATH'}"}
+my $mypath = '/software/components/afsclt';
```

Comment:
also a readonly? (in principal, you can use `$self->prefix()` for this value, but it's used quite a lot, so variable is fine)